modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminDisplayContext.java

```diff
@@ -799,14 +799,22 @@ else if (navigation.equals("mine") || navigation.equals("recent")) {
 	private Hits _getHits(SearchContainer<RepositoryEntry> searchContainer)
 		throws PortalException {
 
+		long searchRepositoryId = ParamUtil.getLong(
+			_httpServletRequest, "searchRepositoryId");
+
+		if (searchRepositoryId != 0) {
+			ThemeDisplay themeDisplay =
+				(ThemeDisplay)_httpServletRequest.getAttribute(
+					WebKeys.THEME_DISPLAY);
+
+			themeDisplay.setScopeGroupId(searchRepositoryId);
```

Comment:
This should only happens if the repository is a group (asset lib or a site):
so something similar to this can be useful:
```
Group group = GroupLocalServiceUtil.fetchGroup(searchRepositoryId);
if (group != null) {
...
```

Reply:
Could you list out these cases that repository can be? It can help me re-evaluate this case.

Reply:
On theory, you can add an external repository, as sharepoint/alfresco/etc... 
this kind of repositories won't be groups. 

I do not know how this work on the search plugin but putting the repository id as group id , on that case, it's not correct.

Reply:
Thank for your explanation. For more sure it will works correctly, I updated code following your explanation to cover those cases.

Reply:
great!